onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -2879,20 +2879,11 @@ def aten_hspmm(mat1: TensorType, mat2: TensorType) -> TensorType:
 def aten_hstack(tensors: Sequence[TTensor]) -> TTensor:
     """hstack(Tensor[] tensors) -> Tensor"""
 
-    # TODO: Due to lack of for loop, we couldn't use at::atleast_1d
-    # and examine the first tensor dim like torch did in their implementation.
-
-    # In PyTorch:
-    # Tensor hstack(TensorList tensors) {
-    #   TORCH_CHECK(!tensors.empty(),
-    #            "hstack expects a non-empty TensorList");
-    #   auto rep = at::atleast_1d(tensors);
-    #   if (rep[0].dim() == 1) {
-    #     return at::cat(rep, 0);
-    #   }
-    #   return at::cat(rep, 1);
-    # }
+    # Use another onnx function
+    tensors = aten_atleast_1d(tensors)
```

Comment:
👀

Reply:
Looks like it works.

Reply:
Yes, it works. The error is something else. For this particular case, if we need to move aten::atleast_1d to a private share node, it would be a whole function...

Reply:
I would then lift the whole function. Just don't call one aten in another